init/luks.go

```diff
@@ -411,3 +440,23 @@ func handleLuksBlockDevice(blk *blkInfo) error {
 
 	return luksOpen(blk.path, m)
 }
+
+func findOrCreateLuksMapping(uuid UUID) *luksMapping {
+	blk := blkInfo{
+		uuid: uuid,
+	}
+
+	for _, o := range luksMappings {
+		if blk.matchesRef(o.ref) {
+			return &o
+		}
+	}
+
+	// didn't locate the device make a new one
+	luksMappings = append(luksMappings, luksMapping{
```

Comment:
this part can be simpler - make a variable for the new luksMapping and then return it
```go
m := luksMapping{
                ref:  &deviceRef{refFsUUID, uuid},
		name: "luks-" + uuid.toString(),
	}
luksMappings = append(luksMappings, m)
return &m
```

Reply:
Good catch, I toyed with that function a bit changing what I wanted to do with it and didn't fully clean it up. I threw it in luks since it is really pretty luks specific, but I do see the utils file as well, as this is also mostly a utility function would it be better suited there? Wasn't sure what you preference was on locality.

Reply:
I have no strong preference, so let's keep it here